### ISSUE TITLE
403 errors after upgrade to v0.39.0

### ISSUE THREAD

**Issue by LukoJy3D** [prymitive/pint-action#26]
After updating to 0.39.0, the pint github action started failing with the error message:
`
level=fatal msg="Execution completed with error(s)" error="submitting reports: failed to create a new check run: POST https://api.github.com/repos/../../check-runs: 403 You must authenticate via a GitHub App. []
`

Do I understand correctly that we need to add an app to the repository with write privileges for pint to work?

**Comment by prymitive** [comment]
I'm using this workflow - https://github.com/prymitive/test-pint/blob/main/.github/workflows/pint.yml - and it works as expected without any app, example here: https://github.com/prymitive/test-pint/pull/11/files.

Did you forget to pass the token or have some extra restrictions on your workflow?

**Comment by LukoJy3D** [comment]
Wow, I did not expect such a fast response! Big thanks for the information. I am afk for the holidays but will get back to this as soon as I come online. Glad to see that this works at your end without any gh app. We have self-hosted runners, so maybe some hiccups in our configuration

**Comment by LukoJy3D** [comment]
Hello, I tried granulating permissions in the workflow, but with no success (even with write-all). Maybe you can share what scopes have you defined for the token you are using?

**Comment by prymitive** [comment]
Digging deeper into checks api docs I see that GitHub App is the required environment, so it might work in my test setup only by pure coincidence or there's some extra knob that I have enabled there without even noticing (?).
Need to do more reading on this before I can fix it.

**Comment by prymitive** [comment]
Are you using a custom token for this?
Can you share you workflow file please? I'm having trouble trying to reproduce it.

**Comment by prymitive** [comment]
Also: is your PR raised from the same repository or from a fork?

**Comment by LukoJy3D** [comment]
> Are you using a custom token for this?

PAT - Classic one, not fine-grained. Current permissions: workflow, write:packages, read:packages, delete:packages 

> Can you share you workflow file please?

Currently locked on 1.4.0, but usually, its on a main branch
```
name: alerts-lint

on:
  pull_request:
    branches:
      - master
    paths:
      - alerts/**

jobs:
  lint:
    runs-on:
      - self-hosted
    steps:
      - name: Checkout
        uses: actions/checkout@v3
        with:
          fetch-depth: 0
      - name: Run pint
        uses: prymitive/pint-action@v1.4.0
        with:
          token: ${{ TOKEN }}
```

> is your PR raised from the same repository or from a fork?

Same repo that have self-hosted runners attached

**Comment by prymitive** [comment]
Could you try with `token: ${{ github.token }}` ?

**Comment by prymitive** [comment]
This should be fixed in pint v0.40.0, please update your workflow to use [v1.7.0](https://github.com/prymitive/pint-action/releases/tag/v1.7.0)
